src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionObjectRpm.hpp

```diff
@@ -39,140 +45,321 @@ class VersionObjectRpm final : public IVersionObject
+     *
+     * This function compares two version strings to determine which version is newer, using comparison rules that
+     * mirror those used by RPM. This ensures compatibility with standard RPM package versioning. The comparison process
+     * involves several steps to accurately compare version strings that may include alphanumeric characters and tildes
+     * (~).
+     *
+     * The process begins by checking for string equality. If both version strings are identical, they are considered
+     * equal. If not, each string is converted into a list of characters for a detailed comparison.
+     *
+     * The comparison involves the following steps:
+     * - First, any leading characters that are not alphanumeric and not tildes (~) are trimmed from both character
+     * lists.
+     * - Next, leading tildes (~) are removed. The presence of a tilde at the beginning of a version indicates that the
+     * version should be considered older than versions without a tilde, even if numerically larger. Therefore, if one
+     * version starts with a tilde and the other does not, the version without the tilde is considered newer.
+     * - The comparison then proceeds by popping and comparing consecutive blocks of digits or letters from the start of
+     * each list. If a difference is found, the comparison result of these blocks is returned immediately.
+     * - If all characters are compared without finding a difference, the comparison concludes by examining the lengths
+     * of any remaining characters in the lists. A version with remaining characters is considered newer, unless those
+     * characters begin with a tilde, in which case it is considered older. This step accounts for scenarios where one
+     * version string has been entirely consumed, leaving characters in the other, such as when comparing "1.05b" to
+     * "1.05".
+     *
+     * @param leftVer The left-hand side version string to compare.
+     * @param rightVer The right-hand side version string to compare.
+     * @return Returns 1 if `leftVer` is considered newer, -1 if `rightVer` is newer, or 0 if both versions are
+     * considered equal.
+     */
+    int compareVersions(const std::string& leftVer, const std::string& rightVer) const
+    {
+        if (leftVer == rightVer)
+        {
+            return LEFT_EQ_RIGHT;
+        }
+
+        std::vector<char> leftSide(leftVer.begin(), leftVer.end());
+        std::vector<char> rightSide(rightVer.begin(), rightVer.end());
 
-            auxChrOne = *lhsStrAux;
-            *lhsStrAux = '\0';
-            auxChrTwo = *rhsStrAux;
-            *rhsStrAux = '\0';
+        while (!leftSide.empty() && !rightSide.empty())
+        {
+            removeNonAlnumAndNonTilde(leftSide);
+            removeNonAlnumAndNonTilde(rightSide);
 
-            if (auxOne == lhsStrAux)
+            if (leftSide.front() == '~' && rightSide.front() == '~')
```

Comment:
This section seems to reflect with more fidelity the algorithm at https://github.com/rpm-software-management/rpm/blob/master/rpmio/rpmvercmp.c 
- Were we missing the `~`  logic?
- Why are we ignoring the `^` section?

Reply:
We are using the  `rpmdev-vercomp` algorithm.

Reply:
Is the link I shared the source you used?

Reply:
Yes :)

Reply:
> Were we missing the ~ logic?

After the tests, now we are considering the `~`

> Why are we ignoring the ^ section?

We might need this. I'd simply copy the required logic instead of taking it to C++ and/or remove sections.